index.js

```diff
@@ -73,58 +74,89 @@ if (argv.channel) {
   client.on('message', function (from, to, message) {
     var op = parse(message)
     if (!op) return
+    var channel = (to === argv.name) ? from : argv.channel
+    var key = op.key
     switch (op.command) {
-      case 'add': return add(new Buffer(op.key, 'hex'))
+      case 'add':
+        pending.push({key: key, channel: channel})
+        return ar.add(new Buffer(key, 'hex'), function (err) {
+          if (err) return sendMessage(err, channel)
+          sendMessage(null, channel, 'Adding ' + key)
+        })
       case 'rm':
-      case 'remove': return remove(new Buffer(op.key, 'hex'))
-      case 'status': return status()
+      case 'remove':
+        pending = pending.filter(function(obj) {
+          return obj.key !== key && obj.feedKey !== key
+        })
+        return ar.remove(new Buffer(key, 'hex'), function (err) {
+          if (err) return sendMessage(err, channel)
+          sendMessage(null, channel, 'Removing ' + key)
+        })
+      case 'status':
+        return status(function (err, msg) {
+          sendMessage(err, channel, msg)
+        })
```

Comment:
could you move the `return` statements until after the functions in the switch cases? its confusing to read since there in a function decl inside

``` js
status(function () {
 ...
})
return
```